framework/include/samplers/Sampler.h

```diff
@@ -111,6 +159,12 @@ class Sampler : public MooseObject, public SetupInterface, public DistributionIn
 
   /// Initial random number seed
   const unsigned int & _seed;
+
+  /// Data offsets for computing location based on global index
+  std::vector<unsigned int> _offsets;
+
+  /// Total number of rows
+  unsigned int _total_rows = 0;
```

Comment:
Yeah, I ran into problems with this the first time I tried this when we first switched. I can't remember what it was now. It's inconsistent with everything else but I suppose it's OK.